Task_1_2_3

MIZINCHIK opened the pull request.

MIZINCHIK committed:
Initial commit for soft deadline

MIZINCHIK committed:
fixed gradlew

MIZINCHIK committed:
fixed wrapper

alradish commented on Task_1_2_3/src/main/java/io/github/mizinchik/GraphImpl.java:
```diff
@@ -0,0 +1,102 @@
+import java.util.NoSuchElementException;
+
+public class GraphImpl<I, L extends Number> implements Graph<GraphImpl<I, L>, EdgeImpl<I, L>, VertexImpl<I, L>, I, L> {
+    private ArrayList<VertexImpl<I, L>> vertices;
```
`List`

alradish reviewed (changes requested):
Нужно разобраться с зависимостями. Старайтесь не создавать циклических зависимостей т.е. 
Вершина ничего не должна знать о графе, и т.д.
При объявлении типов старайтесь использовать общие интерфейсы, например List    и Map

MIZINCHIK committed:
Pre-Javadoc commit

Dependencies fixed.
General interfaces are now used in the interfaces.
Floyd-Warshall algorithm implemented.
IO handled.
Test written.

MIZINCHIK committed:
Javadoc added

MIZINCHIK committed:
Javadog fixes

MIZINCHIK committed:
Javadog fixes

MIZINCHIK committed:
Reviewdog fixes

alradish commented on Task_1_2_3/src/main/java/io/github/mizinchik/EdgeImpl.java:
```diff
@@ -0,0 +1,79 @@
+ */
+public class EdgeImpl<I> implements Edge<VertexImpl<I>, I, Double> {
+    private Double length;
+    private final VertexImpl<I> start;
+    private final VertexImpl<I> end;
```
Используйте интерфейс `Vertext`

alradish commented on Task_1_2_3/src/main/java/io/github/mizinchik/GraphImpl.java:
```diff
@@ -0,0 +1,449 @@
+        VertexImpl<I> startingPoint = extractedVertices.get(startingPointInt);
+        List<Map.Entry<String, Double>> topSort = topSort(startingPoint);
+        for (Map.Entry<String, Double> entry : topSort) {
+            System.out.print(entry.getKey() + "(" + entry.getValue() + ") ");
```
Такому не место в конструкторе

alradish reviewed (changes requested):
- Task_1_2_2 лишнее

MIZINCHIK committed:
Brought Task_1_2_2 back

MIZINCHIK committed:
Requested changes applied

alradish commented on Task_1_2_3/src/main/java/io/github/mizinchik/EdgeImpl.java:
```diff
@@ -0,0 +1,79 @@
+     * @param start vertex where it starts
+     * @param end vertex where it ends
+     */
+    public EdgeImpl(Double length, String name, VertexImpl<I> start, VertexImpl<I> end) {
```
Vertex

alradish commented on Task_1_2_3/src/main/java/io/github/mizinchik/GraphImpl.java:
```diff
@@ -0,0 +1,456 @@
+    private HashMap<String, VertexImpl<I>> namesToVertices;
+    private HashMap<String, ArrayList<EdgeImpl<I>>> edgeOutLists;
+    private HashMap<String, HashMap<String, EdgeImpl<I>>> adjacencyMatrix;
+    private final ArrayList<EdgeImpl<I>> edges;
+    private final ArrayList<String> vertices;
```
List

alradish commented on Task_1_2_3/src/main/java/io/github/mizinchik/GraphImpl.java:
```diff
@@ -0,0 +1,456 @@
+ * @param <I> values stored in vertices
+ */
+public class GraphImpl<I> implements Graph<EdgeImpl<I>, VertexImpl<I>, I, Double> {
+    private HashMap<String, VertexImpl<I>> namesToVertices;
+    private HashMap<String, ArrayList<EdgeImpl<I>>> edgeOutLists;
+    private HashMap<String, HashMap<String, EdgeImpl<I>>> adjacencyMatrix;
```
Map

alradish reviewed (changes requested):
- Если будете на семинаре, напомните мне рассказать про `Vertex` vs `VertexImpl`
- Подойдите на паре по поводу второй лабы 1_2_2. Я не могу залить Task_1_2_3 пока не починится Task_1_2_2

MIZINCHIK committed:
Final fix

MIZINCHIK committed:
Fixed and cleaned code
